pkgs/desktops/gnome-3/3.20/core/libpeas/default.nix

```diff
@@ -1,13 +1,15 @@
 { stdenv, fetchurl, pkgconfig, intltool, gnome3
-, glib, gtk3, gobjectIntrospection, python, pygobject3
+, glib, gtk3, gobjectIntrospection, python3, python3Packages, ncurses, python2, python2Packages
 }:
 
 stdenv.mkDerivation rec {
   inherit (import ./src.nix fetchurl) name src;
 
+  configureFlags = [ "--enable-python3" "--enable-python2" ];
+
   buildInputs =  [
-   intltool pkgconfig glib gtk3 gobjectIntrospection python pygobject3
-   gnome3.defaultIconTheme
+   intltool pkgconfig glib gtk3 gobjectIntrospection python3 python3Packages.pygobject3 python2
+   gnome3.defaultIconTheme ncurses
```

Comment:
to guarantee that `python2` and `python3` packages not mingle it is indeed possible to use a `buildEnv` for each.

Example
```
with import <nixpkgs> {};
with pkgs;

let
  python2 = python2Packages.python.withPackages ( ps: with ps; [ toolz ] );
  python3 = python3Packages.python.withPackages ( ps: with ps; [ numpy ] );
in stdenv.mkDerivation {
  name = "doublepython";
  unpackPhase = "true";
  propagatedBuildInputs = [ python2 python3 ];
}
```
Use `nix-shell --pure` and you can test whether `python3` can import `toolz` and whether `python2` can import `numpy`.

This is something to put in the docs...